src/components/data/TableCell.tsx

```diff
@@ -42,18 +42,25 @@ export default function TableCell({
           // on scroll with sticky headers.
           'text-left border-b border-b-grey-5': isHeadCell,
           'border-t': isHeadCell && !borderless,
-          'border-none': !isHeadCell,
+          'border-none': !grid && !isHeadCell,
           // Apply a very subtle bottom border to the last row in the table (not
           // in the head). This can help delineate the end of data in tables
           // with sparse row data. Only apply border if row is not selected.
           // This uses Tailwind's nested-group syntax. See
           // https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups
           'group-last/unselected:border-b group-last/unselected:border-grey-2 group-last/unselected:border-dotted':
-            !isHeadCell,
+            !grid && !isHeadCell,
+          // We would typically set `border` in cells together with
+          // `border-collapse` in the parent table. However, that doesn't work
+          // for sticky headers, so we mimic the behavior by conditionally
+          // setting individual cell borders.
+          'group-[:not(:last-child)]:border-b': grid,
```

Comment:
Can you clarify why this doesn't work with sticky headers?

Reply:
When the `thead` element is set with `position: sticky` and `border-collapse` is not set as `separate` in the `table`, this happens:

https://github.com/hypothesis/frontend-shared/assets/2719332/b8af99f7-26cd-4c96-a9a4-9419d1c93942

As you can see, the header itself is sticky, but the borders of its cells are not, and they "move up" when scrolling.

I'm not sure if this is a bug or there's a reason for browsers to do this (I'll try to investigate a bit), but it's something Lyza already faced and documented here https://github.com/hypothesis/frontend-shared/blob/main/src/components/data/Table.tsx#L60-L63

In my initial iteration of this PR I decided to ignore that because I didn't fully understand the problem, but once I tested the grid table with a scroll container, I saw it.

The solution applied here fulfills with the requirements documented by Lyza, "A side effect is that borders need to be set primarily on table cells, not rows", by setting a bottom border to all cells except those from the last row (via `group-[:not(:last-child)]:border-b`) and right border to all cells except the last one on every row (via `divide-x` in every row).